test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -2339,10 +2340,61 @@ def forward(self, x):
         m = convert_pt2e(m)
         m(*example_inputs)
 
+    def test_cond_op(self):
+        m = TestHelperModules.Cond()
+        example_inputs = m.example_inputs()
+        m = export_for_training(m, example_inputs).module()
+        quantizer = XNNPACKQuantizer().set_global(
+            get_symmetric_quantization_config(),
+        )
+        gm = prepare_pt2e(m, quantizer)
 
-instantiate_parametrized_tests(TestQuantizePT2E)
+        from torch.ao.quantization.observer import ObserverBase
 
-devices = ["cpu", "cuda"]
-if TEST_HPU:
-    devices.append("hpu")
-instantiate_device_type_tests(TestQuantizePT2E, globals(), only_for=devices)
+        for node in gm.graph.nodes:
+            if node.target == torch.ops.higher_order.cond:
+                true_gm = getattr(gm, node.args[1].target)
+                for n in true_gm.graph.nodes:
+                    if n.target == torch.ops.aten.conv2d.default:
+                        assert n.meta.get("quantization_annotation", None) is not None
+                        input_act = n.args[0]
+                        weight = n.args[1]
+                        output = list(n.users)[0]
+                        for obs_node in [input_act, weight, output]:
+                            obs_ins = getattr(true_gm, obs_node.target)
+                            assert isinstance(obs_ins, ObserverBase)
+                false_gm = getattr(gm, node.args[2].target)
+                for n in false_gm.graph.nodes:
+                    if n.target == torch.ops.aten.conv2d.default:
+                        assert n.meta.get("quantization_annotation", None) is not None
+                        input_act = n.args[0]
+                        weight = n.args[1]
+                        output = list(n.users)[0]
+                        for obs_node in [input_act, weight, output]:
+                            obs_ins = getattr(false_gm, obs_node.target)
+                            assert isinstance(obs_ins, ObserverBase)
+
+        gm(*example_inputs)
```

Comment:
There is a data dependent error when we trace into https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/observer.py#L1281 because this line is testing whether the value of the tensor equal to some value. HistogramObserver class is tracing-friendly. will the model be trained eagerly? We could come up with a way that avoids tracing in we're eagerly running. It will work after that.